builtin/client/readonly.lua

```diff
@@ -0,0 +1,7 @@
+-- Set the global environment to a read-only mode.
+
+local f=function (t,i) error("cannot redefine global variable `"..i.."'",2) end
```

Comment:
You could put this function directly into `setmetatable`.
Also the code style is not consistent with the other builtin code.